### PR TITLE
Desktop notifications using snorenotify

### PR DESCRIPTION
**Pull request by sudden6**
Things to be done:

- [x] add setting to disable desktop notifications
- [ ] build snorenotify on AppImage
- [x] build snorenotify on Flatpak (check if even needed)
- [ ] ~build snorenotify on Windows (check if working)~
- [ ] ~build snorenotify on Mac~
- [ ] ~add snorenotify to simple_make.sh (opensuse has it in the repo)~

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/5227)

### CODE REVIEW

**Comment by sudden6** [comment]
@probonopd I hope it's ok to mention you here, but I have a problem with linuxqtdeploy. I want to include libsnorenotify in the AppImage, but it has dynamically loaded plugins, which `ldd` can't find. Is there a way to to explicitely include them in the AppImage?

EDIT: The relevant build script is: https://github.com/qTox/qTox/pull/5227/commits/8966a54bfb7c01f76684e885757ee17a4731f2c9#diff-4c5c91692db6a9b8b86cb7fb86e3f11c

**Comment by sudden6** [comment]
I consider this ready for testing on Linux and Mac now.

**Comment by Diadlo** [comment]
i3wm. Doesn't work. Maybe related:
```
[22:41:39.689 UTC] :0 : Warning: libpng warning: iCCP: known incorrect sRGB profile
[22:41:39.689 UTC] :0 : Warning: libpng warning: iCCP: cHRM chunk does not match sRGB
```

**Comment by sudden6** [comment]
No, this is not related, there should be something along the lines of `notification backend: ` with a non empty string. Note notifying is turned off by default and and needs to be explicitly enabled in cmake for now.

File: src/platform/desktop_notifications/desktopnotify.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+void DesktopNotify::notifyFriendMessage()
+{
+    const Settings& s = Settings::getInstance();
+    if(!(s.getNotify() && s.getDesktopNotify())) {
+        return;
+    }
+
+    const QString title = tr("New message received");
```
Comment (Diadlo) [inline src/platform/desktop_notifications/desktopnotify.cpp:94]: Maybe add friend name in message?

File: src/platform/desktop_notifications/desktopnotify.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+void DesktopNotify::notifyGroupInvite()
+{
+    const Settings& s = Settings::getInstance();
+    if(!(s.getNotify() && s.getDesktopNotify())) {
+        return;
+    }
+
+    const QString title = tr("New message received");
```
Comment (Diadlo) [inline src/platform/desktop_notifications/desktopnotify.cpp]: "New group invite"?

File: src/platform/desktop_notifications/desktopnotify.cpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+void DesktopNotify::notifyFriendRequest()
+{
+    const Settings& s = Settings::getInstance();
+    if(!(s.getNotify() && s.getDesktopNotify())) {
+        return;
+    }
+
+    const QString title = tr("New message received");
```
Comment (Diadlo) [inline src/platform/desktop_notifications/desktopnotify.cpp]: "New friend request"?

**Review by Diadlo — CHANGES_REQUESTED** [review]
Seems to works

File: src/platform/desktop_notifications/desktopnotify.cpp (hunk 1 above)
Comment (sudden6) [reply src/platform/desktop_notifications/desktopnotify.cpp:94]: I thought about this, but left it out for now due to privacy concerns and ease of implementation.

**Review by Talkless — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5227)* status: 0 of 1 LGTMs obtained

___
*[src/platform/desktop_notifications/desktopnotify.h, line 29 at r2](https://reviewable.io/reviews/qtox/qtox/5227#-LJSqZSF7yxIBpKOhejn:-LJSqZSF7yxIBpKOhejo:b-t4n5p0) ([raw file](https://github.com/qtox/qtox/blob/baf923707b4724121866fda0c43bc6289844dd63/src/platform/desktop_notifications/desktopnotify.h#L29)):*
> ```objc
> private:
>     Snore::SnoreCore& notifyCore;
>     NotificationPtr groupInvite = nullptr;
> ```

{nullptr}

**Review by sudden6 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5227)* status: 0 of 1 LGTMs obtained

___
*[src/platform/desktop_notifications/desktopnotify.h, line 29 at r2](https://reviewable.io/reviews/qtox/qtox/5227#-LJSqZSF7yxIBpKOhejn:-LK8QACzEAMn-pzaeupA:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/baf923707b4724121866fda0c43bc6289844dd63/src/platform/desktop_notifications/desktopnotify.h#L29)):*
<details><summary><i>Previously, Talkless (Vincas Dargis) wrote…</i></summary><blockquote>

{nullptr}
</blockquote></details>

Done.
___
*[src/platform/desktop_notifications/desktopnotify.cpp, line 58 at r1](https://reviewable.io/reviews/qtox/qtox/5227#-LH9i9wq3WRX4SMTVheW-r1-58:-LK8QBaB0agjarV5JGVO:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/37cc2725a4f63835fa67df9b4c0378d9a20b3118/src/platform/desktop_notifications/desktopnotify.cpp#L58)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

"New friend request"?
</blockquote></details>

Done.
___
*[src/platform/desktop_notifications/desktopnotify.cpp, line 76 at r1](https://reviewable.io/reviews/qtox/qtox/5227#-LH9i9wq3WRX4SMTVheW-r1-76:-LK8QCLN743FJ5Va6My_:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/37cc2725a4f63835fa67df9b4c0378d9a20b3118/src/platform/desktop_notifications/desktopnotify.cpp#L76)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

"New group invite"?
</blockquote></details>

Done.

**Comment by sudden6** [comment]
I dropped the support for non Linux systems for now, I couldn't really figure out how to cross compile this library.

**Comment by sudden6** [comment]
Assigned to myself to prevent accidental merge.

This is just for archive purposes of potential MacOS support.

**Pull request closed without merging**